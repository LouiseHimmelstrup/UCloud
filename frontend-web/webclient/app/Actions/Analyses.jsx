import { Cloud } from "../../authentication/SDUCloudObject";
import { SET_LOADING, RECEIVE_ANALYSES, SET_PAGE_SIZE } from "../Reducers/Analyses";

export const fetchAnalyses = (analysesPerPage, currentPage) =>
    Cloud.get(`/hpc/jobs/?itemsPerPage=${analysesPerPage}&page=${currentPage}`)
         .then(({response}) => receiveAnalyses(response))

const receiveAnalyses = ({ items, itemsPerPage, pageNumber, pagesInTotal }) => ({
    type: RECEIVE_ANALYSES,
    analyses: items,
    analysesPerPage: itemsPerPage,
    pageNumber: pageNumber,
    totalPages: pagesInTotal
})

export const setPageSize = (pageSize) => ({
    type: SET_PAGE_SIZE,
    pageSize
});

export const setLoading = (loading) => ({
    type: SET_LOADING,
    loading
});