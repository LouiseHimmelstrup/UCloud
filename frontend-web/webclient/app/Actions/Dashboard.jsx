import { Cloud } from "../../authentication/SDUCloudObject";
import { SET_ALL_LOADING, RECEIVE_FAVORITES, RECEIVE_RECENT_ANALYSES, RECEIVE_RECENT_FILES } from "../Reducers/Dashboard";

export const setAllLoading = (loading) => ({
    type: SET_ALL_LOADING,
    loading
});

export const fetchFavorites = () =>
    Cloud.get(`/files?path=${Cloud.homeFolder}`).then(({ response }) => {
        let actualFavorites = response.filter(file => file.favorited);
        const subsetFavorites = actualFavorites.slice(0, 10);
        return receiveFavorites(subsetFavorites);
    });

export const receiveFavorites = (favorites) => ({
    type: RECEIVE_FAVORITES,
    favorites
});

export const fetchRecentFiles = () =>
    Cloud.get(`/files?path=${Cloud.homeFolder}`).then(({ response }) => {
        const recentSubset = response.slice(0, 10);
        return receiveRecentFiles(recentSubset);
    });

const receiveRecentFiles = (recentFiles) => ({
    type: RECEIVE_RECENT_FILES,
    recentFiles
});

export const fetchRecentAnalyses = () =>
    Cloud.get("/hpc/jobs").then(({ response }) => {
        const recentAnalyses = response.items.slice(0, 10);
        return receiveRecentAnalyses(recentAnalyses);
    });

const receiveRecentAnalyses = (recentAnalyses) => ({
    type: RECEIVE_RECENT_ANALYSES,
    recentAnalyses
});