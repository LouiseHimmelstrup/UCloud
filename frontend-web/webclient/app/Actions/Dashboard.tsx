import { Cloud } from "../../authentication/SDUCloudObject";
import { SET_ALL_LOADING, RECEIVE_FAVORITES, RECEIVE_RECENT_ANALYSES, RECEIVE_RECENT_FILES } from "../Reducers/Dashboard";
import { failureNotification } from "../UtilityFunctions";
import { Analysis, File, SetLoadingAction, Action } from "../types/types";

interface Fetch<T> extends Action { content: T[] }

export const setAllLoading = (loading: boolean): SetLoadingAction => ({
    type: SET_ALL_LOADING,
    loading
});

export const fetchFavorites = ():Promise<Fetch<File>> =>
    Cloud.get(`/files?path=${Cloud.homeFolder}/Favorites`).then(({ response }) =>
        receiveFavorites(response.items.slice(0, 10))
    ).catch(() => {
        failureNotification("Failed to fetch favorites. Please try again later.")
        return receiveFavorites([])
    });

export const receiveFavorites = (content: File[]):Fetch<File> => ({
    type: RECEIVE_FAVORITES,
    content
});

export const fetchRecentFiles = ():Promise<Fetch<File>> =>
    Cloud.get(`/files?path=${Cloud.homeFolder}`).then(({ response }) =>
        receiveRecentFiles(response.items.slice(0, 10))
    ).catch(() => {
        failureNotification("Failed to fetch recent files. Please try again later.");
        return receiveRecentFiles([]);
    });

const receiveRecentFiles = (content):Fetch<File> => ({
    type: RECEIVE_RECENT_FILES,
    content
});

export const fetchRecentAnalyses = ():Promise<Fetch<Analysis>> =>
    Cloud.get(`/hpc/jobs/?itemsPerPage=${10}&page=${0}`).then(({ response }) =>
        receiveRecentAnalyses(response.items)
    ).catch(() => {
        failureNotification("Failed to fetch recent analyses. Please try again later.")
        return receiveRecentAnalyses([]);
    });

const receiveRecentAnalyses = (content: Analysis[]):Fetch<Analysis> => ({
    type: RECEIVE_RECENT_ANALYSES,
    content
});