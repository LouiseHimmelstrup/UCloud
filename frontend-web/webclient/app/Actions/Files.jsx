import {Cloud} from "../../authentication/SDUCloudObject"
import { RECEIVE_FILES, UPDATE_FILES_PER_PAGE, UPDATE_FILES, SET_LOADING, UPDATE_PATH, TO_PAGE } from "../Reducers/Files"

/*
* Creates a promise to fetch files. Sorts the files based on sorting function passed,
* and implicitely sets @filesLoading to false in the reducer when the files are fetched.
*/
export const fetchFiles = (path, sorting, sortAscending) => 
  Cloud.get(`files?path=${path}`).then((res) => {
      let files = res.response;
      files.forEach(file => file.isChecked = false);
      if (sorting) {
        files = sorting(files, sortAscending);
      }
      return receiveFiles(files);
  });

export const toPage = (pageNumber) => ({
  type: TO_PAGE,
  pageNumber: pageNumber
});

/*
* Updates the files stored. 
* Intended for use when sorting the files, checking or favoriting, for instance.
*/
export const updateFiles = (files) => {
  return {
    type: UPDATE_FILES,
    files
  }
};

export const setLoading = (loading) => {
  return {
    type: SET_LOADING,
    loading
  }
};

export const updatePath = (newPath) => ({
    type: UPDATE_PATH,
    path: newPath
});

export const updateFilesPerPage = (filesPerPage, files) => {
  files.forEach(file => file.isChecked = false);
  return {
    type: UPDATE_FILES_PER_PAGE,
    filesPerPage,
    files
  }
};

const receiveFiles = (files, path) => ({
  type: RECEIVE_FILES,
  files,
  path
});