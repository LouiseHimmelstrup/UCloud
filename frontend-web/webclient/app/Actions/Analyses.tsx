import { Cloud } from "../../authentication/SDUCloudObject";
import { failureNotification } from "../UtilityFunctions";
import { SET_ANALYSES_LOADING, RECEIVE_ANALYSES, SET_ANALYSES_PAGE_SIZE } from "../Reducers/Analyses";
import { Page, Analysis } from "../types/types";

export const fetchAnalyses = (analysesPerPage: number, currentPage: number) =>
    Cloud.get(`/hpc/jobs/?itemsPerPage=${analysesPerPage}&page=${currentPage}`)
        .then(({ response }) => receiveAnalyses(response)).catch(() => {
            failureNotification("Retrieval of analyses failed, please try again later.");
            return setLoading(false);
        });

const receiveAnalyses = (page: Page<Analysis>) => ({
    type: RECEIVE_ANALYSES,
    page: page
});

export const setPageSize = (pageSize: number) => ({
    type: SET_ANALYSES_PAGE_SIZE,
    pageSize
});

export const setLoading = (loading: boolean) => ({
    type: SET_ANALYSES_LOADING,
    loading
});