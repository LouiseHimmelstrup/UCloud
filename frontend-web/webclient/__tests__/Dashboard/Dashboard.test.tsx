import * as React from "react";
import Dashboard from "Dashboard/Dashboard";
import { create } from "react-test-renderer";
import * as DashboardActions from "Dashboard/Redux/DashboardActions";
import dashboard from "Dashboard/Redux/DashboardReducer";
import notifications from "Notifications/Redux/NotificationsReducer";
import status from "Navigation/Redux/StatusReducer";
import { Provider } from "react-redux";
import { configureStore } from "Utilities/ReduxUtilities";
import { initNotifications, initDashboard, initStatus, ReduxObject } from "DefaultObjects";
import { mockFiles_SensitivityConfidential } from "../mock/Files";
import { MemoryRouter } from "react-router";
import { analyses } from "../mock/Analyses";
import * as AccountingRedux from "Accounting/Redux";
import { ThemeProvider } from "styled-components";
import { theme } from "ui-components";
import { Store } from "redux";

const createStore = () => {
    return configureStore(
        {
            notifications: initNotifications(),
            dashboard: initDashboard(),
            status: initStatus(),
            ...AccountingRedux.init()
        },
        {
            dashboard,
            notifications,
            status,
            ...AccountingRedux.reducers
        }
    );
};

const WrappedDashboard: React.FunctionComponent<{ store: Store<ReduxObject> }> = props => {
    return <Provider store={props.store}>
        <ThemeProvider theme={theme}>
            <MemoryRouter>
                <Dashboard />
            </MemoryRouter>
        </ThemeProvider>
    </Provider >
};

describe("Dashboard Component", () => {
    test("Mount with favorites", () => {
        const store = createStore();
        store
        store.dispatch(DashboardActions.receiveFavorites(mockFiles_SensitivityConfidential.items));
        expect(create(<WrappedDashboard store={store} />).toJSON()).toMatchSnapshot();
    });

    // FIXME Requires backend support
    test.skip("Mount with favorites, de-favorite single file", () => {
        const store = createStore();
        store.dispatch(DashboardActions.receiveFavorites(mockFiles_SensitivityConfidential.items));
        expect(create(<WrappedDashboard store={store} />).toJSON()).toMatchSnapshot();
    });

    test("Mount with recent files", () => {
        const store = createStore();
        store.dispatch(DashboardActions.receiveRecentFiles(mockFiles_SensitivityConfidential.items));
        const files = store.getState().dashboard.recentFiles;
        files.forEach(it => { it.modifiedAt = 0 });
        store.dispatch(DashboardActions.receiveRecentFiles(files));
        expect(create(<WrappedDashboard store={store} />).toJSON()).toMatchSnapshot();
    });

    test("Mount with recent files", () => {
        const store = createStore();
        store.dispatch(DashboardActions.receiveRecentAnalyses(analyses.items));
        expect(create(<WrappedDashboard store={store} />).toJSON()).toMatchSnapshot();
    });
});