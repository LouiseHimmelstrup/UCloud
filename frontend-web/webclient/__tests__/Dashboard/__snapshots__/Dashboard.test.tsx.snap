// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component Mount with favorites 1`] = `
<div
  className="ui cards mobile-padding"
>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Favorite files
      </div>
      <div
        className="ui large divided list"
        role="list"
      >
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <i
              aria-hidden="true"
              className="blue star icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey image icon"
            />
            <a
              href="/files//home/user@user.telecity/"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-200px"
              >
                Screenshot_2018-08-09 SDU-eScience SDUCloud-1.png
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <i
              aria-hidden="true"
              className="blue star icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/AABA"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-200px"
              >
                AABA
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <i
              aria-hidden="true"
              className="blue star icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/Favorites"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-200px"
              >
                Favorites
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <i
              aria-hidden="true"
              className="blue star icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/Jobs"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-200px"
              >
                Jobs
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <i
              aria-hidden="true"
              className="blue star icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/A folder the sequel"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-200px"
              >
                A folder the sequel
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <i
              aria-hidden="true"
              className="blue star icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/B"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-200px"
              >
                B
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <i
              aria-hidden="true"
              className="blue star icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/rehjoerijhoerijh"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-200px"
              >
                rehjoerijhoerijh
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <i
              aria-hidden="true"
              className="blue star icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/Test"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-200px"
              >
                Test
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <i
              aria-hidden="true"
              className="blue star icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/somefolder-60D6DAD4-DCF1-415E-B4D6-9BCEF028369E"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-200px"
              >
                somefolder-60D6DAD4-DCF1-415E-B4D6-9BCEF028369E
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <i
              aria-hidden="true"
              className="blue star icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/somefolder-24ACCA0C-2FAA-4B6C-8F31-7F1E633A54DB"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-200px"
              >
                somefolder-24ACCA0C-2FAA-4B6C-8F31-7F1E633A54DB
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recently used files
      </div>
      <h3>
        <small>
          No recently used files
        </small>
      </h3>
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recent Jobs
      </div>
      <h3>
        <small>
          No results found
        </small>
      </h3>
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recent notifications
      </div>
      <h3>
        <small>
          No notifications
        </small>
      </h3>
      <div
        className="ui divided list"
        role="list"
      />
    </div>
  </div>
</div>
`;

exports[`Dashboard Component Mount with no data 1`] = `
<div
  className="ui cards mobile-padding"
>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Favorite files
      </div>
      <i
        aria-hidden="true"
        className="circle notched loading icon"
      />
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recently used files
      </div>
      <i
        aria-hidden="true"
        className="circle notched loading icon"
      />
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recent Jobs
      </div>
      <i
        aria-hidden="true"
        className="circle notched loading icon"
      />
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recent notifications
      </div>
      <h3>
        <small>
          No notifications
        </small>
      </h3>
      <div
        className="ui divided list"
        role="list"
      />
    </div>
  </div>
</div>
`;

exports[`Dashboard Component Mount with notifications 1`] = `
<div
  className="ui cards mobile-padding"
>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Favorite files
      </div>
      <i
        aria-hidden="true"
        className="circle notched loading icon"
      />
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recently used files
      </div>
      <i
        aria-hidden="true"
        className="circle notched loading icon"
      />
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recent Jobs
      </div>
      <i
        aria-hidden="true"
        className="circle notched loading icon"
      />
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recent notifications
      </div>
      <div
        className="ui divided list"
        role="list"
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <i
            aria-hidden="true"
            className="blue share alternate icon middle aligned"
          />
          <div
            className="content"
          >
            <div
              className="header"
            >
              Share Request
            </div>
            <div
              className="description"
            >
              jonas@hinchely.dk has shared a file with you
            </div>
          </div>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <i
            aria-hidden="true"
            className="blue share alternate icon middle aligned"
          />
          <div
            className="content"
          >
            <div
              className="header"
            >
              Share Request
            </div>
            <div
              className="description"
            >
              jonas@hinchely.dk has shared a file with you
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dashboard Component Mount with recent files 1`] = `
<div
  className="ui cards mobile-padding"
>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Favorite files
      </div>
      <h3>
        <small>
          No favorites found
        </small>
      </h3>
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recently used files
      </div>
      <div
        className="ui large divided list"
        role="list"
      >
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey image icon"
            />
            <a
              href="/files//home/user@user.telecity/"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-117px"
              >
                Screenshot_2018-08-09 SDU-eScience SDUCloud-1.png
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/AABA"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-117px"
              >
                AABA
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/Favorites"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-117px"
              >
                Favorites
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/Jobs"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-117px"
              >
                Jobs
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/A folder the sequel"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-117px"
              >
                A folder the sequel
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/B"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-117px"
              >
                B
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/rehjoerijhoerijh"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-117px"
              >
                rehjoerijhoerijh
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/Test"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-117px"
              >
                Test
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/somefolder-60D6DAD4-DCF1-415E-B4D6-9BCEF028369E"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-117px"
              >
                somefolder-60D6DAD4-DCF1-415E-B4D6-9BCEF028369E
              </span>
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            <div
              className="description"
            >
              49 years ago
            </div>
          </div>
          <div
            className="content"
          >
            <i
              aria-hidden="true"
              className="grey folder icon"
            />
            <a
              href="/files//home/user@user.telecity/somefolder-24ACCA0C-2FAA-4B6C-8F31-7F1E633A54DB"
              onClick={[Function]}
            >
              <span
                className="limited-width-string-117px"
              >
                somefolder-24ACCA0C-2FAA-4B6C-8F31-7F1E633A54DB
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recent Jobs
      </div>
      <h3>
        <small>
          No results found
        </small>
      </h3>
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recent notifications
      </div>
      <h3>
        <small>
          No notifications
        </small>
      </h3>
      <div
        className="ui divided list"
        role="list"
      />
    </div>
  </div>
</div>
`;

exports[`Dashboard Component Mount with recent files 2`] = `
<div
  className="ui cards mobile-padding"
>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Favorite files
      </div>
      <h3>
        <small>
          No favorites found
        </small>
      </h3>
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recently used files
      </div>
      <h3>
        <small>
          No recently used files
        </small>
      </h3>
      <div
        className="ui large divided list"
        role="list"
      />
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recent Jobs
      </div>
      <div
        className="ui large divided list"
        role="list"
      >
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            Failure
          </div>
          <i
            aria-hidden="true"
            className="red x icon"
          />
          <div
            className="content"
          >
            <a
              href="/applications/results/bbcf5395-e78e-4fa7-a4e1-989fe5ce21ee"
              onClick={[Function]}
            >
              figlet-count
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            Failure
          </div>
          <i
            aria-hidden="true"
            className="red x icon"
          />
          <div
            className="content"
          >
            <a
              href="/applications/results/15274bec-ef72-4ed2-97ff-2be1829a2db1"
              onClick={[Function]}
            >
              figlet
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            Failure
          </div>
          <i
            aria-hidden="true"
            className="red x icon"
          />
          <div
            className="content"
          >
            <a
              href="/applications/results/3544b177-e9a7-4323-bcde-c4d6e442dc36"
              onClick={[Function]}
            >
              figlet
            </a>
          </div>
        </div>
        <div
          className="item itemPadding"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="right floated content"
          >
            Success
          </div>
          <i
            aria-hidden="true"
            className="green check icon"
          />
          <div
            className="content"
          >
            <a
              href="/applications/results/e28defb0-d66c-4461-91f6-b09afd85480c"
              onClick={[Function]}
            >
              figlet-count
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Recent notifications
      </div>
      <h3>
        <small>
          No notifications
        </small>
      </h3>
      <div
        className="ui divided list"
        role="list"
      />
    </div>
  </div>
</div>
`;
